raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -383,6 +390,100 @@ def handle_transferreroute(
     return TransitionResult(payment_state, events)
 
 
+def handle_transfer_reroute_light(
+    payment_state: InitiatorPaymentState,
+    state_change: ActionTransferRerouteLight,
+    channel_identifiers_to_channels: ChannelMap,
+    storage: SerializedSQLiteStorage
+) -> TransitionResult[InitiatorPaymentState]:
+
+    print("raiden/transfer/mediated_transfer/initiator_manager.py:396 running handle_transfer_reroute_light")
+
+    refund_transfer = state_change.refund_transfer
+
+    if not refund_transfer:
+        raise RaidenUnrecoverableError("Refund Transfer not found, state has changed with action "
+                                       "ActionTransferRerouteLight but refund_transfer param was not set")
+
+    store_refund_transfer = StoreMessageEvent(refund_transfer.message_identifier,
+                                              refund_transfer.payment_identifier,
+                                              1,
+                                              refund_transfer,
+                                              True,
+                                              LightClientProtocolMessageType.PaymentRefund,
+                                              refund_transfer.recipient)
+
+    try:
+        initiator_state = payment_state.initiator_transfers.get(state_change.transfer.lock.secrethash)
+        channel_identifier = initiator_state.channel_identifier
+        channel_state = channel_identifiers_to_channels[initiator_state.transfer.initiator].get(channel_identifier)
+    except KeyError:
+        return TransitionResult(payment_state, list())
+
+    refund_transfer = state_change.transfer
+    original_transfer = initiator_state.transfer
+
+    is_valid_lock = (
+        refund_transfer.lock.secrethash == original_transfer.lock.secrethash
+        and refund_transfer.lock.amount == original_transfer.lock.amount
+        and refund_transfer.lock.expiration == original_transfer.lock.expiration
+    )
+
+    is_valid_refund = channel.refund_transfer_matches_transfer(refund_transfer, original_transfer)
+    is_valid, channel_events, _, _ = channel.handle_receive_lockedtransfer_light(
+        channel_state, refund_transfer, storage
+    )
+
+    if not is_valid_lock or not is_valid_refund or not is_valid:
+        return TransitionResult(payment_state, list())
+
+    events: List[Event] = [store_refund_transfer]
+    events.extend(channel_events)
+
+    return TransitionResult(payment_state, events)
+
+
+def handle_lock_expired_light(
+    payment_state: InitiatorPaymentState,
```

Comment:
If this method wasnt implemented we need to think about p2p expirations.

What happened BEFORE this PR if we have this scenario?

1. LC wants to send a payment to FN
2. LC sends the payment but the hub goes offline or the lc stop polling
3. the payment expires
4. the partner sends a LE

Reply:
Lets talk with @debrayperalta and make a code walktrough of that scenario.

Reply:
I'm not really sure about what happen in that case, in this case i know for sure that it wasn't handled right and that's why it wasn't working. We can review it with him sure.

Reply:
was this reviewed?

Reply:
Not yet

Reply:
We changed this already as we spoke with @marcosmartinez7 in a meeting, i don't know if we want to review this with @debrayperalta anyway, please confirm @marcosmartinez7 when you can, thanks!